janus_core/helpers/janus_types.py

```diff
@@ -75,6 +75,15 @@ class PostProcessKwargs(TypedDict, total=False):
     vaf_output_file: Optional[PathLike]
 
 
+class CorrelationKwargs(TypedDict, total=False):
+    """Arguments for on-the-fly correlations."""
+
+    # Pairs of indexed observables, ('s_xy', 's_xy')
+    correlations: Sequence[tuple[str, str]]
+    # Blocks, points, window, and update frequency
+    correlation_parameters: Sequence[tuple[int, int, int, int]]
```

Comment:
Alongside the discussion on the actual names of these keys below,

I was thinking something like this so all values are clearer,

```
class CorrelationKwargs(TypedDict, total=False):
    observable_a: str
    observable_b: str
    blocks: int
    points: int
    window: int
    update_frequency: int
```

and

```
correlation_kwargs: Optional[Sequence[CorrelationKwargs]]
```

but then these are just a sequence of args I guess.

Not sure what else to write other than "observable_a". The idea is its calculating $\langle A(t)B(t+\tau)\rangle$. It could just be a.